modules/gapi/test/gapi_async_test.cpp

```diff
@@ -262,13 +341,57 @@ TYPED_TEST_P(stress, test){
 }
 REGISTER_TYPED_TEST_CASE_P(stress, test);
 
+template<typename case_t>
+struct cancel : ::testing::Test{};
+TYPED_TEST_CASE_P(cancel);
+
+TYPED_TEST_P(cancel, basic){
+    cancel_struct cancel_struct_ {100};
+    std::size_t num_tasks = 100;
+    std::vector<TypeParam> requests; requests.reserve(num_tasks);
+
+    for (auto i = num_tasks; i>0; i--){
+        requests.emplace_back(&cancel_struct_);
+    }
+    for (auto&& r : requests){
+        //first request will cancel other on it's execution
+        r.start_async(cancel_struct_.ctx, r.in_args(), r.out_args());
+    }
+
+    size_t canceled = 0 ;
+    for (auto&& r : requests){
+        try {
+            r.wait_for_result();
+        }catch (cv::gapi::wip::GAsyncCanceled& e){
+//        }catch (std::exception& e){
+            //operation was canceled
+            std::cout<< "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n"
+                    <<e.what() << std::endl;
+            ++canceled;
+        }catch (...)
+        {
+
+        }
+    }
+    ASSERT_GT(canceled, size_t(0));
```

Comment:
Just want to check that I'm understanding this corectly. Should it be not just greater than 0? Should it be equal to 99?

Reply:
tend to agree: `ASSERT_EQ(canceled, num_tasks - 1)`? is such behavior guaranteed?

Reply:
There are some pecularities in multithread cases. As far as I understand (num_tasks - num_threads) is the most correct answer.

Reply:
in the __current__ implementation (with a single background thread) it will be 99 - but it is an implementation detail. 

So best we can do is to make sure that at least some requests were cancelled

Reply:
`size_t(0)` or `size_t(0u)`?